ghost-cat-app/src/app/View/image-thumbnails-page/image-thumbnails-page.component.ts

```diff
@@ -167,6 +172,72 @@ export class ImageThumbnailsPageComponent implements OnInit {
   public selectedBoundingBox(bb: BoundingBoxModel): void {
     this.sidebarComponent.selectedBoxChanged(bb);
   }
+
+  @ViewChild('image') imageDetailsComponent: ImageDetailsComponent;
+
+  public deletedBoundingBoxes(bb: BoundingBoxModel[]) {
+    for (let i = 0; i < this.items.length; ++i) {
+      let currBoxes = this.items[i].data.boundingBoxes;
+      let filteredBoxes = [];
+
+      for (let j = 0; j < currBoxes.length; ++j) {
+        if (bb.find(b => b.id == currBoxes[j].id) == undefined) {
+          filteredBoxes.push(currBoxes[j]);
+        }
+      }
+
+      this.items[i].data.boundingBoxes = filteredBoxes;
+    }
+  }
+
+  public addNewBox(src: string) {
+    if (src == null) {
+      this.imageDetailsComponent.addNewBoundingBox(false);
+      this.newBBLookup = null;
+    } else {
+      let myId = uuid.v4();
+      var parts = src.split("/");
+      var imgId = parts[parts.length - 1];
+      let newBBModel: BoundingBoxModel = null;
+
+      let item = this.items.find(item => item.data.src == src);
+      if (item != undefined) {
+        let classes = { "Mule Deer": 0, "Cow": 0, "Sheep": 0, "Other": 0 };
```

Comment:
I hard coded this just as a temporary thing... Does anyone know how to like create an object based on the classes in the database?

Reply:
I'm pretty sure that all of the classes would be in `item.data.animalLabels` at this point, but they are stored as an array of strings. I'd have to think a little bit harder about how you might be able to play around with the types here to make that useful

Reply:
Yeah I tried to get it from animalLabels but I couldn't figure out how to use those to make a custom object. I tried some stuff but eventually just decided to use this as a temporary solution